### ISSUE TITLE
Process Tanner Internal Patient Data

### ISSUE THREAD

**Issue by stratabuild** [stratasan/wrike-test#147]
##### 
##### Please leave your contact information
**Name**
Alex Antonison

**Email**
alex@stratasan.com

**Type of request**
Support (Presales, client launch, non-critical items)

**Who is this request for?**
Tanner

**What is the request?**
Process Tanner's Internal Patient Data into our system.

**Why is the request needed?**
Internal Patient Data has been sold to Tanner and we are obligated to support this.

**How often do you hear this request?**

**When is this request needed by?**
Urgent - within 4 weeks

**Request For**
Other

**Add screenshots or other files to help**

**Links to queries/reports**
### Urgent Request - Need w/in 4 weeks
**Why is this an urgent request?**
This is urgent since this is a contractual obligation to Tanner

**Since this is an urgent request, please alert the product team about it in the #product slack channel. Did you do that?**

**What date is this needed by?**
2020-03-28

┆Issue is synchronized with this [Wrike Task](https://www.wrike.com/open.htm?id=477071657)
┆Attachments: <a href="https://share.unito.io/XT0MNxuaP8QmIK-qc5hTXYqRIGijzDVN6nPcQu4zVdfZ">Tanner Test File Discussion.eml</a>

**Comment by stratabuild** [comment]
➤ Christian commented:

did this get done?

**Comment by stratabuild** [comment]
➤ Beth Duke commented:

Christian Lawrence  this should not be canceled. They sent us the data. We sent back feedback. We are waiting on their response.

**Comment by stratabuild** [comment]
➤ Beth Duke commented:

On secong thought... Is this a duplicate of https://www.wrike.com/open.htm?id=477081745

**Comment by stratabuild** [comment]
➤ Beth Duke commented:

Christian Lawrence

**Comment by stratabuild** [comment]
➤ Beth Duke commented:

EIther way we still need to process their data ocne they answer our feedback 😀

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

see this ticket: https://www.wrike.com/open.htm?id=477081745

**Comment by stratabuild** [comment]
➤ Dave Sellers commented:

Yes as soon as we get a file format from them, along with all applicable crosswalks which map new values to the values received in prior extracts, we will load what they have sent.  The ability to map new values to old ones is required for instances where they want to continue using trending reports which were developed to help them evaluate their marketing effectiveness.

An response from Tanner was provided to me on 04/02

**Comment by stratabuild** [comment]
➤ alex-strat commented:

This is held up as we are waiting for Tanner to provide Stratasan the correct data for processing.

**Comment by stratabuild** [comment]
➤ Morgan Atkins commented:

waiting on tanner to provide dataa

**Comment by stratabuild** [comment]
➤ Christian commented:

@alex-strat when we look at our backlog of ETL work that we reviewed yesterday, where would this be slotted in so we can see the impact.

**Comment by stratabuild** [comment]
➤ Christian commented:

also, why is this a data science ticket and not a data engineering one?

**Comment by stratabuild** [comment]
➤ Beth Duke commented:

I am still pending an update from Tanner but the specific hold up at this time is related to: The physician information is now enhanced to have both a numeric ID and name field, whereas the old data contained only a 5-character identifier for the physician.  mapping required so that physician information can be trended across both the old and new datasets in a consistent manner. They do not have mapping for this and are trying to determine if it is possible. They believe it is a massive undertaking but my Tanner contacts are pushing their internal contact to make this happen. Two of their recent requests would have been better with the 2019Q4 internal data so I know its high on their list but Covid messes with priorities for them.

**Comment by stratabuild** [comment]
➤ alex-strat commented:

@clawdyne This is a Data Science ticket vs Data Engineering as @drs1832 and @var-ela have been processing Internal Patient Data so they have existing scripts and processes to support this.  Since I believe the entire Internal Patient Data product requires a complete re-work to focus on improving both the client experience in getting us the data as well as figure out ways to reduce the burden of processing the data, I opted to leave this as is until that re-work happens.

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

Beth Duke date will continue to move and we can't guarantee a delivery date until we get their data.

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

Beth Duke circling back on this item and request. Any news?

**Comment by stratabuild** [comment]
➤ Beth Duke commented:

Christian Lawrence I met with Melintha and Martin at Tanner today. however,  I have not heard from Christie Lambert at Tanner since 05/08/20. Melintha was planning to reach out to her today. It is my understanding its a pretty big ask for them to resolve the physician mapping but its very important in order to see trending information. The last two projects completed definitely would have been better with the 2019Q4 data so I feel like Melintha will start pushing them harder for a resolution and new file to us. Hopefully more details coming soon.

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

As discussed with Beth Duke , we are cancelling this ticket and will wait for you to tell us to reengage.

**Comment by stratabuild** [comment]
➤ Beth Duke commented:

Alex Antonison Tanner is about to send the crosswalks and a test file.  They are asking if one month of data is a good test file? (2000 records)

**Comment by stratabuild** [comment]
➤ alex-strat commented:

@drs1832 Can you speak to the sample crosswalk file?

**Comment by stratabuild** [comment]
➤ Dave Sellers commented:

A data file of that size and the necessary crosswalks should be fine to validate improvements on the feedback we had raised last exchange.  The more records the better as far as the sample data file size goes, but if 2000 is all they can do for now that's fine.

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

Beth Duke any update?

**Comment by stratabuild** [comment]
➤ Dave Sellers commented:

Based on review of latest test file provided on 06/09:

File format
in the past have had problems with inconsistent delimiters (comma, pipe, etc.).  Latest file supplied was .XLSX so mostly just need to land on a consistent file format to be used.  Preference is pipe-delimited text file

Crosswalks/legacy equivalencies
Patient ID: In latest file they provided an appropriate "legacy MRN" field to map to the old-format patient identifiers
Facility ID: still need this
Patient Race: still need this
Admit Source: need confirmation that this is the same set as the CMS standard Admit Source descriptions
Discharge Disposition: need confirmation that this is the same set as the CMS standard Admit Source descriptions
Patient Type: still need this
Financial Class: still need this
Physician: With the latest file, they provided an appropriate crosswalk to physician ID's in the new system to the physician ID's in the legacy system

DRG information
data files from previous system was using MS-DRGs.  New data files, including latest, are including what appear to be APR-DRG's.  This would prevent the ability to trend by DRG code (but could still trend by Product/Sub-Product Line

Diagnosis sequencing
noted issue where there were values in position #3 of the 5 Dx fields, but nothing in #1 and 2, which seemed counterintuitive.  Seems to have been corrected in latest file

**Comment by stratabuild** [comment]
➤ Beth Duke commented:

Christian Lawrence I was unexpectedly out two days last week with a family emergency. I have emailed Tanner today with Dave's feedback. I have my normal monthly call on Thursday with them but believe I will get a response prior from Tanner.

**Comment by stratabuild** [comment]
➤ Beth Duke commented:

I have confirmation that they will send MS DRG so that trending is still possible.

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

Beth Duke can we place this on hold?

**Comment by stratabuild** [comment]
➤ Christian Lawrence commented:

closing and moving to data engineering

**Comment by stratabuild** [comment]
➤ Beth Duke commented:

Christian Lawrence We met with Tanner today and They are sending us another test file on Wednesday of next week
